addons/mail/controllers/discuss.py

```diff
@@ -138,15 +138,16 @@ def mail_message_post(self, thread_model, thread_id, post_data, **kwargs):
         return thread.message_post(**{key: value for key, value in post_data.items() if key in allowed_params}).message_format()[0]
 
     @http.route('/mail/message/update_content', methods=['POST'], type='json', auth='public')
-    def mail_message_update_content(self, message_id, body):
+    def mail_message_update_content(self, message_id, body, attachment_ids):
         guest = request.env['mail.guest']._get_guest_from_request(request)
         message_sudo = guest.env['mail.message'].browse(message_id).sudo().exists()
         if not message_sudo.is_current_user_or_guest_author and not guest.env.user.has_group('base.group_system'):
             raise NotFound()
-        message_sudo._update_content(body=body)
+        message_sudo._update_content(body=body, attachment_ids=attachment_ids)
         return {
             'id': message_sudo.id,
             'body': message_sudo.body,
+            'attachments': [('insert', request.env['ir.attachment'].browse(attachment_ids)._attachment_format())],
```

Comment:
Use `env` from `message_sudo` and not from `request` or it will fail for guest.
Also don't trust only the `attachment_ids` param for what to return, directly access the attachments actually on `message_sudo` instead (similar to what is done to body).